### PR TITLE
TE 489 fix derived package for gradle

### PR DESCRIPTION
**Pull request by gunther-bachmann**
please inspect this [pull request](https://github.com/test-editor/test-editor-xtext/pull/304) first

### CODE REVIEW

File: common/org.testeditor.dsl.common/src/org/testeditor/dsl/common/util/classpath/ClasspathUtil.xtend [hunk 1]
```diff
@@ -44,22 +44,38 @@ class ClasspathUtil {
 			path = new Path(path.toFile.absolutePath)
 			cpEntry = path.getBuildToolClasspathEntry
 		}
+		if (cpEntry === null) {
+			logger.error("Could not find corresponding classpath entry for path='{}'.", path)
+			return 'com.example'
```
Comment (franzbecker) [inline common/org.testeditor.dsl.common/src/org/testeditor/dsl/common/util/classpath/ClasspathUtil.xtend]: Not sure about the `com.example` here. Maybe the default / empty package instead?

File: common/org.testeditor.dsl.common/src/org/testeditor/dsl/common/util/classpath/ClasspathUtil.xtend [hunk 2]
```diff
@@ -44,22 +44,38 @@ class ClasspathUtil {
...
+	}
 
 	def IPath getBuildToolClasspathEntry(IPath path) {
 		logger.info("Searching classpath for {}.", path)
 		val baseDir = path.getBuildProjectBaseDir
-		if (baseDir.toFile.list.contains("pom.xml")) {
-			return mavenClasspathUtil.getMavenClasspathEntries(baseDir).filter[it.isPrefixOf(path)].head
+		switch (baseDir) {
+			case baseDir.isMavenProjectBaseDir: return mavenClasspathUtil.getMavenClasspathEntries(baseDir).findFirst[
+				logger.info("Checking maven classpath='{}'.", it.toOSString)
```
Comment (franzbecker) [inline common/org.testeditor.dsl.common/src/org/testeditor/dsl/common/util/classpath/ClasspathUtil.xtend]: Should be `debug`

File: common/org.testeditor.dsl.common/src/org/testeditor/dsl/common/util/classpath/GradleClasspathUtil.xtend [hunk 3]
```diff
@@ -12,41 +12,50 @@
...
-			result.add(new Path(prjDir + "/src/" + it.substring(it.indexOf("'") + 1, it.lastIndexOf("'")) + "/java"))
-		]
-		return result;
+	/**
+	 * try to extract source set paths from output: line format "sourceSet: 'PATH'"
+	 */
+	@VisibleForTesting
+	protected def List<IPath> parseGradleSourcePaths(String output) {
+		logger.info("Parsing gradle task output printProperties to extract source set.")
+		val lines = output.split(System.getProperty("line.separator"))
```
Comment (franzbecker) [inline common/org.testeditor.dsl.common/src/org/testeditor/dsl/common/util/classpath/GradleClasspathUtil.xtend]: I prefer `CharSource.wrap(output).readLines` or `output.split('\r?\n')`.

File: common/org.testeditor.dsl.common/src/org/testeditor/dsl/common/util/classpath/ClasspathUtil.xtend (hunk 1 above)
Comment (gunther-bachmann) [inline common/org.testeditor.dsl.common/src/org/testeditor/dsl/common/util/classpath/ClasspathUtil.xtend]: default package currently poses a problem we want to adress in a different ticket [OTA-767](http://jira.system.local/browse/OTA-767)

**Pull request merged**